### ISSUE TITLE
Migrate STAC endpoint search to `pystac-client`

### ISSUE THREAD

**Issue by leothomas** [NASA-IMPACT/veda-documentation#7]
From @vincentsarago: 
> sat-search / sat-stac is unmaintained (by us :homer: ) it would be better not to use it

> better use https://github.com/stac-utils/pystac-client

`sat-stac` also causes a dependency incompatibility as it relies on `python-dateutil~=2.7.5`, but `pandas 1.4.3`, which is required by `rioxarray` requires `python-dateutil>2=2.8.1`

**Comment by leothomas** [comment]
Addressed in #16

**Comment by aboydnw** [comment]
@leothomas should we close this as not planned if https://github.com/NASA-IMPACT/veda-documentation/issues/16 covers the same issue?

**Comment by leothomas** [comment]
Oh my bad - I got mixed up. This issue is to migrate the python library `satsearch` to `pystac_client`, as `satsearch` is un-maintained (by us) and was causing dependency issues with `pandoc`. This has been completed/addressed/done in PR #17 , so we can close _this_ issue once #17 is merged. 

#16 is closely related, but not quite the same issue. The notebook referenced in #16 uses the `requests` package to search the API, which is technically a valid approach but it would just be good to streamline all the notebooks into a single way of doing things. I left that one open because it's a lower priority and not blocking anything else, unlike _this issue_ which is blocking the implementation of a CI/CD pipeline (done in #18)

To summarize: we can close this as done once #17 is merged

**Comment by leothomas** [comment]
Addressed in #17
